### ISSUE TITLE
Cross-compiler polution in dependency check

### ISSUE THREAD

**Issue by serbanmaerean** [spack/spack#2927]
I encountered the following situation trying to build netlib-lapack with a new compiler definition and using cmake built w/gcc.

I first build cmake with gcc by itself.  Such build can be used with any other package that uses cmake.  I then go and build netlib-lapack with xl_r and specify ^cmake%gcc as a dependency.  Instead of using the existing cmake build, spack builds cmake again. I see the following when I try to remove cmake:

```
$ spack uninstall cmake
==> Error: cmake matches multiple packages:

-- linux-rhel7-ppc64le / gcc@4.8.5 ------------------------------
yz4andp cmake@3.7.2%gcc~doc+ncurses+openssl+ownlibs~qt

gvispho cmake@3.7.2%gcc fflags="-qzerosize" ~doc+ncurses+openssl+ownlibs~qt


==> Error: You can either:
    a) Use a more specific spec, or
    b) use spack uninstall -a to uninstall ALL matching specs.

```
The fflags="qzerosize" come from the xl_r compiler.

Why does spack wants to build another cmake with the gcc compiler and labels it with the compiler flags from another compiler that has nothing to do w/gcc?  How can I enforce spack to use the cmake version I built the first time when I did target build (as opposed to dependency build)?

Thanks for helping.

**Comment by adamjstewart** [comment]
Duplicate of #2786

**Comment by alalazo** [comment]
This should have been solved by #3379. Please reopen if it is not the case.
